src/devices/imhotepcreation.ts

```diff
@@ -0,0 +1,158 @@
+            tz.thermostat_occupied_cooling_setpoint, tz.thermostat_min_heat_setpoint_limit, tz.thermostat_max_heat_setpoint_limit,
+            tz.thermostat_min_cool_setpoint_limit, tz.thermostat_max_cool_setpoint_limit, tz.thermostat_setpoint_raise_lower],
+        meta: {multiEndpoint: true},
+        endpoint: (device) => {
+            return {
+                'l1': 1, 'l2': 2, 'l3': 3, 'l4': 4, 'l5': 5,
+                'l6': 6, 'l7': 7, 'l8': 8, 'l9': 9, 'l10': 10,
+                'l11': 11, 'l12': 12, 'l13': 13, 'l14': 14, 'l15': 15,
+                'l16': 16,
+            };
+        },
+        exposes: (device, options) => {
+            const features = [];
+
+            if (typeof device !== 'undefined' && (device != null) && device.endpoints) {
+                for (let i = 1; i <= 16; i++) {
+                    const endpoint = device?.getEndpoint(i);
+                    if (endpoint !== undefined) {
+                        const epName = `l${i}`;
+                        features.push(e.enum('system mode', ea.ALL, ['off', 'cool', 'heat']).withProperty('system_mode')
+                            .withEndpoint(epName).withDescription('Thermostat ' + i + ' mode (Off, Heating or Cooling)'));
+                        features.push(e.local_temperature().withEndpoint(epName));
+                        features.push(e.climate().withSetpoint('occupied_heating_setpoint', 5, 30, 0.5, ea.ALL).withEndpoint(epName));
+                        features.push(e.numeric('min_heat_setpoint_limit', ea.ALL).withUnit('°C')
+                            .withDescription('Minimum Heating set point limit')
+                            .withValueMin(5).withValueMax(30).withValueStep(0.5).withEndpoint(epName));
+                        features.push(e.numeric('max_heat_setpoint_limit', ea.ALL).withUnit('°C')
+                            .withDescription('Maximum Heating set point limit')
+                            .withValueMin(5).withValueMax(30).withValueStep(0.5).withEndpoint(epName));
+                        features.push(e.climate().withSetpoint('occupied_cooling_setpoint', 5, 38, 0.5, ea.ALL).withEndpoint(epName));
+                        features.push(e.numeric('min_cool_setpoint_limit', ea.ALL).withUnit('°C')
+                            .withDescription('Minimum Cooling point limit')
+                            .withValueMin(5).withValueMax(38).withValueStep(0.5).withEndpoint(epName));
+                        features.push(e.numeric('max_cool_setpoint_limit', ea.ALL).withUnit('°C')
+                            .withDescription('Maximum Cooling set point limit')
+                            .withValueMin(5).withValueMax(38).withValueStep(0.5).withEndpoint(epName));
+                    }
+                }
+            }
+
+            features.push(e.linkquality());
+
+            return features;
+        },
+        configure: async (device, coordinatorEndpoint, logger) => {
+            if (typeof device !== 'undefined') {
```

Comment:
this check is not needed.

Reply:
Thanks for this feedback, we have removed this check